### PR TITLE
Logcollector integration tests T1: Only future events for macos format

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|#1329 |

## Description

This PR is part of epic issue #1329. It adds a new test for `only-future-events` option for _unified logging system_  format(`macos`).

## Implementation

It uses defined tools for macOS logs creation, using `authd` and `remoted` simulator to gather sent logs.

### MacOS

logcollector_macos

**Build**: https://ci.wazuh.info/job/Test_integration/4960/console

## Documentation

macos_docu_2

Also, this PR fixes archlinux vulnerability detector tests documentation
docu_arch_fix

### CODE REVIEW

**Review by spothound — APPROVED** [review]

File: docs/tests/integration/test_logcollector/test_macos/index.md [hunk 1]
```diff
@@ -13,12 +13,13 @@ Confirm that logcollector works correctly for unified logging system events in m
...
 
 ## List of tests
 
 - **[Test macos format basic](test_macos_format_basic.md)**: Check if `wazuh-logcollector` gather corrrectly generated 
 unified logging system events.
 
-  
\ No newline at end of file
+- **[Test macos format only future events](test_macos_format_only_future_events.md)**: Check if `only-future-events`
+  works correctly for macos log format.
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/index.md:25]:
```suggestion
  works correctly for macOS log format.
```

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Test only future events for macos format
+
+## Overview 
+
+Check if `only-future-events` option for `logcollector` works correctly for macos unified logging 
+system format (ULS).
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md:6]:
```suggestion
Check if `only-future-events` option for `logcollector` works correctly for macOS Unified Logging 
System format (ULS).
```

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Test only future events for macos format
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md:1]:
```suggestion
# Test only future events for macOS format
```

File: docs/tests/integration/test_logcollector/test_macos/index.md [hunk 4]
```diff
@@ -13,12 +13,13 @@ Confirm that logcollector works correctly for unified logging system events in m
...
 
 
 ## List of tests
 
 - **[Test macos format basic](test_macos_format_basic.md)**: Check if `wazuh-logcollector` gather corrrectly generated 
 unified logging system events.
 
-  
\ No newline at end of file
+- **[Test macos format only future events](test_macos_format_only_future_events.md)**: Check if `only-future-events`
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/index.md:24]:
```suggestion
- **[Test macOS format only future events](test_macos_format_only_future_events.md)**: Check if `only-future-events`
```

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+## Overview 
+
+Check if `only-future-events` option for `logcollector` works correctly for macos unified logging 
+system format (ULS).
+
+## Objective
+
+- To confirm that `logcollector` gather all events from ULS when `only-future-events` is disabled
+- To confirm that `logcollector` does not gather old events when `only-future-events` is enabled
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md:11]:
```suggestion
- To confirm that `logcollector` gathers all the events from ULS when `only-future-events` is disabled.
- To confirm that `logcollector` does not gather old events when `only-future-events` is enabled.
```

File: tests/integration/test_logcollector/test_macos/test_macos_format_basic.py [hunk 6]
```diff
@@ -5,13 +5,14 @@
...
 import wazuh_testing.logcollector as logcollector
 from wazuh_testing.tools.configuration import load_wazuh_configurations
 from wazuh_testing.remote import check_agent_received_message
-
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools import LOG_FILE_PATH
+from wazuh_testing.tools.monitoring import FileMonitor
 # Marks
-pytestmark = pytest.mark.tier(level=1)
+pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=1)]
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_macos/test_macos_format_basic.py:15]:
```suggestion
pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=0)]
```
If this is a basic test, it should belong to the 0 tier.

File: tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+import pytest
+from wazuh_testing import logcollector
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.remote import check_agent_received_message
+from wazuh_testing.tools.services import control_service
+
+# Marks
+
+pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=1)]
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.py:15]: We should add this one to the 0 tier IMO.

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: docs/tests/integration/test_logcollector/test_macos/index.md (hunk 1 above)
Comment (Rebits) [reply docs/tests/integration/test_logcollector/test_macos/index.md]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72

File: tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.py (hunk 7 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.py]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72

File: tests/integration/test_logcollector/test_macos/test_macos_format_basic.py (hunk 6 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_macos/test_macos_format_basic.py]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md (hunk 5 above)
Comment (Rebits) [reply docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72

File: docs/tests/integration/test_logcollector/test_macos/index.md (hunk 4 above)
Comment (Rebits) [reply docs/tests/integration/test_logcollector/test_macos/index.md]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md (hunk 3 above)
Comment (Rebits) [reply docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md (hunk 2 above)
Comment (Rebits) [reply docs/tests/integration/test_logcollector/test_macos/test_macos_format_only_future_events.md]: Done 58f4bb663ce1da50f8b24241a5ba741cebcffe72, 0896a7c6d880e70e80b1a243aaad51a8b7a992a3

**Review by BraulioV — APPROVED** [review]

**Pull request merged**